src/Administration/Resources/app/administration/src/module/sw-settings-snippet/page/sw-settings-snippet-list/index.js

```diff
@@ -499,82 +498,84 @@ export default {
 
         onSelectionChanged(selection) {
             this.snippetSelection = selection;
-            this.hasResetableItems = Object.keys(selection).length === 0;
+            this.hasResetableItems = selection && Object.keys(selection).length !== 0;
+        },
+
+        onCloseDeleteModal() {
+            this.showDeleteModal = false;
+            this.modalDeleteAll = false;
+            this.hasResetableItems = false;
+            this.resetItems = [];
         },
 
         onConfirmReset(fullSelection) {
```

Comment:
Could you add some tests for this?

Reply:
Probably not before my vacation. I'll try my best.